app/scripts/metamask-controller.js

```diff
@@ -8890,4 +8899,63 @@ export default class MetamaskController extends EventEmitter {
       initRequest,
     });
   }
+
+  /**
+   * Upgrades an account to support EIP-7702 delegation.
+   * Uses shared EIP-7702 utility to avoid code duplication.
+   *
+   * @param {string} address - The account address to upgrade
+   * @param {string} upgradeContractAddress - The contract address to delegate to
+   * @param {number} chainId - The chain ID for the upgrade
+   * @returns {Promise<{transactionHash: string, delegatedTo: string}>}
+   */
+  async upgradeAccount(address, upgradeContractAddress, chainId) {
+    // Get the network client for the specified chain
+    const networkClientId = this.networkController.findNetworkClientIdByChainId(
+      toHex(chainId),
+    );
+
+    return createEIP7702UpgradeTransaction(
+      {
+        address,
+        upgradeContractAddress,
+        networkClientId,
+      },
+      async (transactionParams, options) => {
+        const transactionMeta = await addTransaction(
+          this.getAddTransactionRequest({
+            transactionParams,
+            transactionOptions: {
+              ...options,
+              origin: 'metamask',
+              requireApproval: true,
+            },
+            waitForSubmit: true,
+          }),
+        );
+        return transactionMeta;
+      },
+    );
+  }
+
+  /**
+   * Gets the current chain ID.
+   *
+   * @returns {number}
+   */
+  getCurrentChainId() {
+    const { selectedNetworkClientId } = this.networkController.state;
```

Comment:
In this context, what is the `selectedNetworkClientId`? Is it the origin making the request?

I see other cases calling `getCurrentChainIdForDomain` with the request origin. Is that functionaly different to this?